application/src/main/java/org/togetherjava/tjbot/commands/free/ChannelStatus.java

```diff
@@ -0,0 +1,206 @@
+     * <p>
+     * This method is called in multithreaded context, however the value is not expected to change
+     * regularly and will not break anything if it is incorrect for a read or two, and it should be
+     * updated before use, which will happen in the using thread. So it has not been made thread
+     * safe.
+     *
+     * @param guild the {@link Guild} that the channel belongs to, to retrieve its name from.
+     * @throws IllegalArgumentException if the guild has not been added, see
+     *         {@link ChannelMonitor#addChannelForStatus(TextChannel)}
+     * @throws IllegalStateException if a channel was added, see
+     *         {@link ChannelMonitor#addChannelToMonitor(long)}, that is not a {@link TextChannel}.
+     *         Since addChannelToMonitor does not access the {@link JDA} the entry can only be
+     *         validated before use instead of on addition.
+     */
+    public void updateChannelName(@NotNull final Guild guild) {
+        GuildChannel channel = guild.getGuildChannelById(channelId);
+        if (channel == null) {
+            throw new IllegalArgumentException(
+                    "The guild passed in '%s' is not related to the channel this status is for: %s"
+                        .formatted(guild.getName(), this));
+        }
+        if (!(channel instanceof TextChannel textChannel)) {
+            throw new IllegalStateException("This channel status was created with the id for a"
+                    + "non-text-channel and status cannot be monitored: '%s'".formatted(channelId));
+        } else {
+            setName(textChannel.getAsMention());
+        }
+    }
+
+    /**
+     * Method to set the channel status to busy, a user id is passed in to keep track of the current
+     * user requesting help. This id will be used to confirm that the author is satisfied with the
+     * channel being marked as free.
+     * <p>
+     * This functionality is not yet implemented so the id can be anything atm. Also note that on
+     * reboot the bot does not currently search for the author so the first time its marked as free
+     * there will be no confirmation.
+     * 
+     * @param userId the id of the user who changed the status to 'busy'
+     */
+    public synchronized void setBusy(final long userId) {
+        if (status.isFree()) {
+            status = ChannelStatusType.BUSY;
+            this.userId = userId;
+        }
+    }
+
+    /**
+     * Method to set the channel status to free, the user id of the previous help requester is not
+     * overwritten by this method. So until another user changes the status to busy the old value
+     * will remain.
+     * <p>
+     * The value will be 0 until the first time that the status is changed from free to busy.
+     * <p>
+     * This functionality is not yet implemented so the id can be anything atm.
+     */
+    public synchronized void setFree() {
+        status = ChannelStatusType.FREE;
```

Comment:
It'd make sense to clear the user ID imo

Reply:
that would require storing id as an optional inside the class itself, rather than having optional as a return type

it will also prevent the OP from continuing with his question after a helper/somebody else closes it

also: since it seems like the next version of the free command will be thread based,
i'm not even sure if we will need to keep the ID of the OP for the bot to be able to interact with him/her. and also with a thread based system, the Statusses dont have to be generated on boot, and there wont be a fixed number of them anymore.

Reply:
> that would require storing id as an optional inside the class itself, rather than having optional as a return type

Don't understand this either.

> it will also prevent the OP from continuing with his question after a helper/somebody else closes it

This doesn't make a lot of sense, it sounds like no-one is able to ask questions after someone runs `/free`?

We shouldn't lie to anyone, if the channel is free, no-one is asking a question right?
So no ID should be stored? But currently there is, and if we get that ID we think someone is asking a question.

But that's not the case, the question has been answered, so the user should be cleared.


If you can't do that, you really need to explain why.

Reply:
> >    that would require storing id as an optional inside the class itself, rather than having optional as a return type
>
> Don't understand this either.

longs can not be null, we can set the id to `0L` but in all honesty thats just guessing/hoping that there wont be a user id of `0L` since it is a valid value in the Snowflake but thats just adding additional Edge cases.
So to actually clear the value, It would either have to be stored as `Long` or `OptionalLong`
(or i would need to introduce a new boolean `isUserIdSet`)
...... which we currently have in the form of isChannelFree



> >  it will also prevent the OP from continuing with his question after a helper/somebody else closes it
>
> This doesn't make a lot of sense, it sounds like no-one is able to ask questions after someone runs /free?

No, this means that anybody at any point for any arbitrary reason can mark a channel as free. Regardless of whether the OP is actually done or not.

And by holding onto the Id the system would have been able to send the user an interaction asking if he agrees with this action or not (Future plans, not implemented yet) and if he doesnt agree it could remove the free status and related messages and still have the id of the OP.

> We shouldn't lie to anyone, if the channel is free, no-one is asking a question right?

We are not lying to anybody, the channel is marked as free and is considered free, the Id is for internal use, its never shown to a user for any reason ever, so no lies.

Reply:
> since it is a valid value in the Snowflake 
It's not, where did you get that information from?


> And by holding onto the Id the system would have been able to send the user an interaction asking if he agrees with this action or not (Future plans, not implemented yet) 
That explains

> We are not lying to anybody, the channel is marked as free and is considered free, the Id is for internal use, its never shown to a user for any reason ever, so no lies.
It should for sure be cleared of the user agrees though, but that's in the future.

Reply:
> > since it is a valid value in the Snowflake
>
> It's not, where did you get that information from?

The starting bits that include the time prefix does not exclude the starting date it reads from .... we are long past that point in time, and are a long way away from the time when it will overflow and be `0L` again but the definition of a Snowflake is is not defined as:

> `0L` UNDEFINED
> for other values:
>    bits 31-24 time prefix ......

> It should for sure be cleared of the user agrees though, but that's in the future.
since it sounds like the next version will be designed to use threads, all of this will probably change


Do you want it set to `0L` on free for this PR?

Reply:
> The starting bits that include the time prefix does not exclude the starting date it reads from .... we are long past that point in time, and are a long way away from the time when it will overflow and be 0L again but the definition of a Snowflake is is not defined as:

See the official doc's on [Snowflakes](
https://discord.com/developers/docs/reference#snowflakes)

I've no idea what that quote means.
But to my understanding it's always the same length, see below.
Thus 0 would be too short since it's missing the `Internal worker ID` the `Internal process ID`  and the `Increment` 
`111111111111111111111111111111111111111111 11111 11111 111111111111`

> Do you want it set to 0L on free for this PR?
You can keep it like this